ctru-rs/src/services/cam.rs

```diff
@@ -0,0 +1,899 @@
+use crate::services::gspgpu::FramebufferFormat;
+use bitflags::bitflags;
+use ctru_sys::Handle;
+use std::cell::RefCell;
+use std::time::Duration;
+
+#[non_exhaustive]
+pub struct Cam {
+    pub inner_cam: RefCell<InwardCam>,
```

Comment:
Are you sure these should be `RefCell`s? We use `RefCell` in `gfx` because in that case the two screens act as separate objects, but in this case I don’t think one can use multiple types of cameras at once. Maybe they should be just held by the struct, and the borrow rules will handle the safety automatically.

Reply:
I think I ended up using them because I saw what was being done in gfx. It made thing easier but I know it's not necessarily the right thing to do. I switched back to just the structs and enclosed setting the camera settings in the example so the mutable borrow is dropped before the loop. Otherwise, the shutter sound wouldn't be able to play since it takes an immutable reference.